packages/language-support/src/parserWrapper.ts

```diff
@@ -267,6 +287,79 @@ class VariableCollector implements ParseTreeListener {
   }
 }
 
+// This listener collects all functions
+class FunctionCollector extends ParseTreeListener {
+  public functions: ParsedFunction[] = [];
+  private tokens: Token[];
+
+  constructor(tokens: Token[]) {
+    super();
+    this.tokens = tokens;
+  }
+
+  enterEveryRule() {
+    /* no-op */
+  }
+  visitTerminal() {
+    /* no-op */
+  }
+  visitErrorNode() {
+    /* no-op */
+  }
+
+  exitEveryRule(ctx: unknown) {
+    if (ctx instanceof FunctionNameContext) {
+      const functionName = this.getFunctionName(ctx);
+
+      const startTokenIndex = ctx.start.tokenIndex;
+      const stopTokenIndex = ctx.stop.tokenIndex;
+
+      const rawText = this.tokens
+        .slice(startTokenIndex, stopTokenIndex + 1)
+        .map((token) => {
+          return token.text;
+        })
+        .join('');
+
+      this.functions.push({
+        name: functionName,
+        rawText: rawText,
+        line: ctx.start.line,
+        column: ctx.start.column,
+        offsets: {
+          start: ctx.start.start,
+          end: ctx.stop.stop + 1,
+        },
+      });
+    }
+  }
+
+  private getFunctionName(ctx: FunctionNameContext): string {
```

Comment:
Suggestion for making the code a bit more understandable: rename this to `getNormalizedFunctionName`